l3kernel/l3tl.dtx

```diff
@@ -1419,6 +1419,22 @@
 %   \end{texnote}
 % \end{function}
 %
+% \begin{function}[added = 2025-06-16]{\tl_rescan_dynamic:n, \tl_rescan_dynamic:V}
+%   \begin{syntax}
+%     \cs{tl_rescan_dynamic:n} \Arg{tokens}
+%   \end{syntax}
+%   Retokenizes the \meta{tokens} by applying the current category code
+%   régime. In contrast to \cs{tl_rescan:nn}, this function retokenizes
+%   the input \enquote{dynamically}: category code changes which occur
+%   whilst processing the retokenized material \emph{will} apply to later
```

Comment:
I don't see how catcode changes can happen while processing. The code below shows everything is first turned into a string, ie catcode 12 throughout and then \scantokens is applied to that, so where is this "dynamic stuff" happening?

Reply:
ah, so not enough context visiblefor me, the stuff is not assigned to a tl but placed into the input stream (should have realized this from the signature. Anyway, that still makes me feel that the names are wrong (but also the original "rescan") The new one is, as you suggested really a retokenize and the one called _rescan is really _recatcode because nothing is really scanned anew there but instead all the catcodes are changes and *then* it is processed while with the new one we have retokenization while (re)processing it